engine/gamesys/src/gamesys/components/comp_private.cpp

```diff
@@ -570,6 +569,25 @@ int AreRenderConstantsUpdated(HComponentRenderConstants constants)
     return constants->m_Updated ? 1 : 0;
 }
 
+// TODO: Can this be improved? The "SetRenderConstant" call feels extremely expensive.
+void CopyRenderConstants(HComponentRenderConstants dst, HComponentRenderConstants src)
```

Comment:
I think we would like to support constant buffers as a graphics resource (similar to textures) on the render object. But before that, we should take a stab att refactoring the render object so it can support arbitrary render resources I think. For example, the "max 8 textures" is a hard limit right now which is very limiting :(